nmagent/client.go

```diff
@@ -284,6 +288,70 @@ func (c *Client) GetHomeAz(ctx context.Context) (AzResponse, error) {
 	return homeAzResponse, nil
 }
 
+func (c *Client) RefreshSecondaryIPsIfNeeded(ctx context.Context) (refreshNeeded bool, ips []string, err error) {
+	if time.Since(c.secondaryIPLastRefreshTime) < c.secondaryIPQueryInterval {
+		return false, nil, nil
+	}
+
+	c.secondaryIPLastRefreshTime = time.Now()
+	res, err := c.getSecondaryIPs(ctx)
+	return true, res, err
+}
```

Comment:
This needs to be done in a wrapper of the client because the ethos of this client is that it will always do what you want it to, not sometimes.

What is the design motivation here? There are better ways to respect rate limits if that's what we're going for.